src/main.rs

```diff
@@ -54,13 +54,7 @@ fn run() -> Result<(), MainError> {
             let manifest_path = cwd.join("package.json");
             let manifest = manifest_path
                 .pipe(File::open)
-                .map_err(|err| {
-                    if err.kind() == ErrorKind::NotFound {
-                        MainError::Pn(PnError::NoPkgManifest { dir: cwd.clone() })
-                    } else {
-                        MainError::from_dyn(err)
-                    }
-                })?
+                .map_err(|err| handle_no_package_manifest_error(err, cwd.clone()))?
                 .pipe(serde_json::de::from_reader::<_, NodeManifest>)
                 .map_err(MainError::from_dyn)?;
```

Comment:
I meant reading and parsing the file in a new function

Reply:
working on it.